### PR TITLE
Added Cucumber Testing Specific Information

### PR DESCRIPTION
**Pull request by LeoNerdoG**
I have added cucumber specifics for kapua project. All of the information
is added in qa.md file, as I did not want to add a new file.
There is basic information regarding testing and after that some
specific information for potential developers/QA people that would want to
contribute.

**Related Issue**
This PR fixes issue #2648.

**Description of the solution adopted**
I have added documentation regarding cucumber integration testing in /docs/developer/guide/qa.md.

**Screenshots**
N/A

**Any side note on the changes made**
There are no side notes that need explanation since this is just a simpel documentation describing cucumber integration tests.

Signed-off-by: Leonardo Gaube <leonardo.gaube@comtrade.com>

### CODE REVIEW

File: docs/developer-guide/en/qa.md [hunk 1]
```diff
@@ -1,28 +1,305 @@
...
 
-This chapter describes a quality assurance process of Kapua. Whenever you commit changes to the `develop` branch, be sure that you have followed those steps:
+This chapter describes a quality assurance process of Kapua. Before you commit changes to the `develop` branch, be sure that you have followed those steps:
 
-## Build tests
+1. Run `mvn clean install -DskipTests -Pconsole,docker` and then `mvn clean install` command to see if all the tests pass correctly
+2. Push changes to you remote repository and wait for the Travis to complete successfully (there should be no test failures, CI environments should always be green):
+	- Travis CI  [![Build](https://api.travis-ci.org/eclipse/kapua.svg)](https://travis-ci.org/eclipse/kapua/) 
+	- Eclipse Hudson [![Hudson](https://img.shields.io/jenkins/s/https/hudson.eclipse.org/kapua/job/Develop.svg)](https://hudson.eclipse.org/kapua/)
+3. Test OpenShift distribution: You should also verify that OpenShift regression test suite works as expected.
```
Comment (Coduz) [inline docs/developer-guide/en/qa.md]: Haven't we removed the qa-openshift module?

File: docs/developer-guide/en/qa.md [hunk 2]
```diff
@@ -1,28 +1,305 @@
...
-- Eclipse Hudson [![Hudson](https://img.shields.io/jenkins/s/https/hudson.eclipse.org/kapua/job/Develop.svg)](https://hudson.eclipse.org/kapua/)
+### Prerequisites (Java, Maven, Cucumber)
+If you are already using Kapua, you can jump straight to section 4), otherwise follow the steps 1), 2) and also 3). These steps are also described in other documents (and Internet), but we are adding them nevertheless. 
 
-## Testing OpenShift distribution
+1) Download And Install IDE (This tutorial uses IntelliJ)
 
-You should also verify that OpenShift regression test suite works as expected. First of all, start local OpenShift cluster as described 
-[here](https://github.com/eclipse/kapua/blob/develop/docs/developer-guide/en/running.md#openshift).
+Go to [IntelliJ website](https://www.jetbrains.com/idea/download/#section=mac) and download the Community Edition IntelliJ. After the installation, we have to install some plugins also, but not before we install Java and Maven.
```
Comment (Coduz) [inline docs/developer-guide/en/qa.md]: Maybe remove the `#section=mac` and make just point to the download page

File: docs/developer-guide/en/qa.md [hunk 3]
```diff
@@ -1,28 +1,305 @@
...
+Now that we have IDE and JDK, we still need Maven. Go to [official Maven website](https://maven.apache.org/download.cgi), select the proper package (it can either ZIP archive or tar.gz archive). Click on download link and save the file (to e.g. Desktop) and after that unzip it to a specific folder in the directory structure (you should never move or delete this maven file once is in this place). An example path is "/opt/apache-maven-3.5.0." 
+When the file is in place, we have to add its path to "PATH" environmental variable. Open Terminal and type: "export PATH=$PATH:~/opt/apache-maven-3.5.0"
+Verify that maven is in place with "mvn -v" command in terminal.
+
+4) Configuring Cucumber (pom.xml files, plugins)
+
+When you have your IDE, Java and Maven, it is time to setup plugins. As said in the beginning, this tutorial uses IntelliJ, but this can all be achieved on Eclipse IDE also (plugin names are a bit different).
+In IntelliJ, go to Preferences -> Plugins -> Marketplace. Search an install the following plugins:
+ - Cucumber for Java
+ - Gherkin (**Do not install "Substep IntelliJ plugin" as it will interfere with other two and Cucumber syntax will not be recognized.**)
```
Comment (Coduz) [inline docs/developer-guide/en/qa.md]: OT this could be also a nice bug report for them :D

File: docs/developer-guide/en/qa.md [hunk 4]
```diff
@@ -1,28 +1,305 @@
...
+ After this is achieved, you have everything that you need to start writing your own tests.
+
+5) Cloning Kapua project
+
+Final step is to clone Kapua from GitHub repository. This is done with command "git clone https://github.com/eclipse/kapua.git ".
+
+### Testing In General
+
+#### Unit testing
+Unit testing is simpe testing, that tests small chunks of code (for example user creation), that do not use other services (permissions, devices...). These tests are tipically done by developers as they know the functionality best.
```
Comment (Coduz) [inline docs/developer-guide/en/qa.md]: `simpe` typo

**Comment by LeoNerdoG** [comment]
Hey @Coduz,
I have commited and pushed the changes. Please merge when tests complete.

**Pull request merged**